### ISSUE TITLE
Copay not recieving funds from coinbase

### ISSUE THREAD

**Issue by Tdildo23** [bitpay/copay#6596]
So I purchased on the copay app through coinbase some BTC today. Once again the amount appeared in my coinbase wallet but not my copay wallet. I recieved a messaeg stating "two step verifcation code required to complete this request. re-play the request with cb2fa token header"

I'm not sure what this means. I then went and disabled coinbase 2fa with the verifcation code, but how do I "replay the request" so that I can get my BTC on copay without getting hit with a transaction fee from coinbase?

**Comment by cmgustavo** [comment]
Hi, 

Unfortunately, you can't replay the request automatically. To get your bitcoin on Copay, you should send it manually from Coinbase app/website.

Copay doesn't support buy/sell throught Coinbase with 2FA enabled. So, the next time make sure to have 2FA disabled before to purchase bitcoin.

Regarding the transaction fee, only to take into account, if you buy bitcoin from Copay integration, also you have to pay fee for the transaction. For example, if your purchase is for 10usd, the total amount to pay could be a little more (Fee from Coinbase purchase, if it applies. And the bitcoin transaction fee from Coinbase to Copay).

**Comment by Tdildo23** [comment]
thanks for the response.

So I'll have to pay the "transaction fee" from coinbase to copay this time?

If I disable the 2fa, will the transaction only be 1 fee? There is currently a small purchase fee will that be the only fee if I don't have to resend the funs?

Or is your last comment saying there are two fees.

If so then what is the point of integrating coinbase and copay if it's no different than sending money between them as two different wallets?

**Comment by cmgustavo** [comment]
> So I'll have to pay the "transaction fee" from coinbase to copay this time?

Yes, if you create a transaction from Coinbase, you have to pay the transaction fee (as every bitcoin transactions).

> If I disable the 2fa, will the transaction only be 1 fee? There is currently a small purchase fee will that be the only fee if I don't have to resend the funs?

> Or is your last comment saying there are two fees.

I don't know if Coinbase charges a fee for purchase with bank account/debit/credit, but I think so. Check this site: https://support.coinbase.com/customer/en/portal/articles/2109597-buy-sell-bank-transfer-fees

As well as you paid fee for purchase, you have to paid the network fee for the transaction (if you want to withdraw your bitcoins from Coinbase). Doing that from Copay, should be automatically, but in your case (2FA enabled and error), you should pay only the bank fee. Otherwise, all users should pay for network fee.

Please, check this site: https://support.coinbase.com/customer/en/portal/articles/815435-does-coinbase-pay-bitcoin-miner-fees-

> Coinbase does not pay the network mining fee on behalf of customers for outgoing digital currency transactions.

Before March 21st, 2017, users didn't pay fee for withdraw bitcoin. Today it 's mandatory.

In summary, using our Coinbase integration does the process to buy/sell faster.

**Comment by Tdildo23** [comment]
Ok I understand better now, so Copay really doesn't offer any real benefits
that any other wallet would then? because you buy and sell the bitcoins
like any other wallet/exchange. The only benefit is if you drop all of your
security to zero you can make one transaction instead of two (yet still pay
the two transaction fee) ok thank you this is much clearer to me now.
